src/lib/utils.js

```diff
@@ -0,0 +1,8 @@
+const transformUserInput = (item) =>
+  item
+    .replace(/[^a-zA-Z ]/g, '')
+    .replace(/\s\s+/g, ' ')
+    .toLowerCase()
+    .trim();
+
+export default transformUserInput;
```

Comment:
Good job on adding the `trim` and `toLowerCase` :+1: . For the regex, you can also[ use `/[\W_]/g` which finds all non word characters and underscores ](https://stackoverflow.com/questions/35324229/how-to-interpret-this-regular-expression-w-g). 
The issue here is that you the third requirement of the issue isn't satisfied:
**"The user’s original input should be what gets saved in the database".**
So, for example, if I want to save an item named "potato!", I should be allowed to save it exactly as it is, no modification should be done to the input (well, except for trimming maybe). But if I try to add "Potato" I shouldn't be able to add it :slightly_smiling_face: 
To do this, you should only compare the user's input to the existing items, without actually modifying the input.